### PR TITLE
Feature: validate HTTP headers

### PR DESCRIPTION
**Pull request by mem**
Validate the headers specified in the HTTP settings.

Signed-off-by: Marcelo E. Magallon <marcelo.magallon@grafana.com>

### CODE REVIEW

**Review by rdubrock — APPROVED** [review]
🦕

File: pkg/pb/synthetic_monitoring/checks_extra.go [hunk 1]
```diff
@@ -23,6 +23,8 @@ import (
 	"net/url"
 	"strconv"
 	"strings"
+
+	"golang.org/x/net/http/httpguts"
```
Comment (electron0zero) [inline pkg/pb/synthetic_monitoring/checks_extra.go:27]: looking at package docs and noticed.

> This package is shared by the standard library (which vendors it) and x/net/http2. It comes with no API stability promise. 

let's surface the fact that any update in httpguts can be a breaking change.

File: pkg/pb/synthetic_monitoring/checks_extra_test.go [hunk 2]
```diff
@@ -728,6 +728,55 @@ func TestValidateLabel(t *testing.T) {
...
+			},
+			expectError: false,
+		},
+		"empty": {
+			input: HttpSettings{
+				Headers: []string{""},
+			},
+			expectError: true,
+		},
+		"no colon": {
```
Comment (electron0zero) [inline pkg/pb/synthetic_monitoring/checks_extra_test.go:758]: let's add a tests for:
- only colon with space around it (`" : "`)
- unicode (emojis or non english text) in header values (`header: यूनिकोड टेक्स्ट`)

> यूनिकोड टेक्स्ट says `Unicode text` in Hindi language

**Review by electron0zero — APPROVED** [review]

File: pkg/pb/synthetic_monitoring/checks_extra.go (hunk 1 above)
Comment (mem) [reply pkg/pb/synthetic_monitoring/checks_extra.go:27]: We have quite a number of packages that fit that fit that description. The whole space under `golang.org/x` for example, as well as BBE.

**Pull request merged**